java-checks/src/main/java/org/sonar/java/filters/LombokFilter.java

```diff
@@ -116,4 +121,42 @@ private static boolean usesAnnotation(ClassTree classTree, List<String> annotati
     }
     return false;
   }
+
+  private boolean generatesPrivateConstructor(ClassTree classTree) {
+    if (usesAnnotation(classTree, GENERATE_PRIVATE_CONSTRUCTOR)) {
+      return true;
+    }
+    SymbolMetadata metadata = classTree.symbol().metadata();
+    for (String annotation : GENERATE_CONSTRUCTOR) {
```

Comment:
You can replace this part by :
```
return GENERATE_CONSTRUCTOR.stream().anyMatch(metadata -> metadata.isAnnotatedWith(annotation) && generatesPrivateAccess(metadata, annotation));
```

I would also move the `isAnnotatedWith(annotation)`directly in the `gernatePrivateAccess` method, so it makes the stream simpler.